zeroconf.py

```diff
@@ -1082,7 +1082,11 @@ def __init__(self, type, name, address=None, port=None, weight=0,
                     bytes for the text field)
         server: fully qualified name for service host (defaults to name)"""
 
-        if not name.endswith(type):
+        compare_type = type
+        if '_sub' in type:
```

Comment:
Thanks, I implemented it with `._sub` since `_sub.` could match something like `_sub._foo._tcp.local` not sure if this would be legal though.